htlcswitch/resolve.go

```diff
@@ -0,0 +1,187 @@
+package htlcswitch
+
+import (
+	"github.com/go-errors/errors"
+	"github.com/jessevdk/go-flags"
+	"github.com/lightningnetwork/lnd/lnwallet"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"os"
+	"path/filepath"
+	"time"
+
+	"encoding/hex"
+	pb "github.com/ExchangeUnion/swap-resolver/swapresolver"
+)
+
+const (
+	defaultConfigFilename = "resolve.conf"
+	defaultServerAddress  = "127.0.0.1:10000"
+	defaultTls            = false
+)
+
+type config struct {
+	Tls                bool `long:"tls" description:"If tls should be used or not"`
+	CaFile             string
+	ServerAddr         string `long:"serveraddr" description:"host and port of the resolver"`
+	ServerHostOverride string
+}
+
+var (
+	cfg = &config{
+		Tls:                defaultTls,
+		CaFile:             "",
+		ServerAddr:         defaultServerAddress,
+		ServerHostOverride: "",
+	}
+)
+
+func isResolverActive() bool {
+	// first see if we have a configuration file at the working directory. If
+	// we miss that, the resolver is not active
```

Comment:
Assuming the resolver config file merges with the main config file, how are we going to check if it's active? Will it be another config option? `resolver.active` or something like that?

Reply:
Yes. Added a TODO for it.